cli/Dockerfile

```diff
@@ -17,11 +17,10 @@ RUN export DEBIAN_FRONTEND=noninteractive \
      # az cli
      && curl -sSL https://aka.ms/InstallAzureCLIDeb | bash \
      # nodejs
-     && curl -fsSL https://deb.nodesource.com/setup_18.x | bash - \
-     && apt-get install -y nodejs \
+     #&& curl -fsSL https://deb.nodesource.com/setup_18.x | bash - \
+     && apt-get install -y nodejs gcc g++ make npm\
      # python
-     && apt-get install -y --no-install-recommends python3-pip python-dev python3-venv \
-     && python3 -m pip install --upgrade pip \
+     && apt-get install -y --no-install-recommends python3-pip python3-venv \
```

Comment:
Version differences: 

| Tool | Latest container version | Version in PR | 
| ---- | ------------------------ | ------------- | 
| Debian | 11 | 12 | 
| dotnet | 6.0.417 | 8.0.100 | 
| node | 18.17.1 | 18.13.0 | 
| npm | 9.6.7 | 9.2.0 | 
| python | 3.9.2 | 3.11.2 | 
| gcc/g++ | n/a | 12.2.0 | 
| make | n/a | 4.3 | 

* Linux distro upgrade -- This should not be a major concern as it is the base image for the .NET container. Looks like 8.0 is not shipping on MCR for `bullseye` (aka Debian 11). OK ✅
* `dotnet` -- This is the latest released version and comes with the container. OK ✅
* `node`/`npm` -- These differ only by minor version which won't impact `azd` per se but in the case of node/npm the versions are going backward. Do we care? If so, use instructions here for onboarding later versions: https://github.com/nodesource/distributions#installation-instructions
* `python` -- Minor version upgrade to Python. Not expected to cause problems as we do our template dev container testing on Python 3.10 which is one minor version ahead.